### PR TITLE
Throw error on unknown arguments

### PR DESCRIPTION
**Pull request by cideM**
This should address issue #37 

`commandLineArgs` already recognizes unknown flags, so if this PR is expected behavior, there is not a whole lot that needed implementing.

### CODE REVIEW

**Comment by cideM** [comment]
Hm seems `commandLineArgs` also considers e.g., files passed as arguments as unknown. I'll work on it.

**Comment by cideM** [comment]
Updated the PR to parse the unknown arguments so only actual flags are considered.

File: argv.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
 const fs = require("fs");
 const path = require("path");
 const commandLineArgs = require("command-line-args");
+const pluralize = require("pluralize");
```
Comment (CapacitorSet) [inline argv.js]: I'd rather not add trivial dependencies. A simple "Unknown arguments: ${flags}" will do.

File: argv.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
 
 const flags = JSON.parse(fs.readFileSync(path.join(__dirname, "flags.json"), "utf8"));
 
+function isFlag(argument) {
+	// e.g., --test and -t should match
+	return /^(--|-)\w*$/.test(argument);
+}
+
+function throwIfUnknownFlag(unknownArguments) {
+	const flags = unknownArguments ? unknownArguments.filter(isFlag) : [];
```
Comment (CapacitorSet) [inline argv.js]: The conditional can be replaced with default arguments in ES6: `function throwIfUnknownFlag(unknownArguments = []) { ...`

File: argv.js [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+
+	if (flags.length) {
```
Comment (CapacitorSet) [inline argv.js]: There is a cleaner way to do this: `if (unknownArguments.some(isFlag))`.

File: argv.js (hunk 2 above)
Comment (cideM) [reply argv.js]: How I did not see this is beyond me, thanks

File: argv.js (hunk 1 above)
Comment (cideM) [reply argv.js]: Cool, will throw it out

**Pull request merged**